File: backend/src/main/java/io/github/artemy/osipov/shop/exception/EntityNotFoundException.java
package io.github.artemy.osipov.shop.exception;

import lombok.Getter;

@Getter
public class EntityNotFoundException extends RuntimeException {
    private final Class<?> targetClass;
    private final Object id;

    public EntityNotFoundException(Class<?> targetClass, Object id) {
        super(generateMessage(targetClass, id));
        this.targetClass = targetClass;
        this.id = id;
    }

    private static String generateMessage(Class<?> targetClass, Object id) {
        return String.format(
                "Entity '%s' with id '%s' not exists",
                targetClass.getSimpleName(), id
        );
    }
}

File: backend/src/main/kotlin/io/github/artemy/osipov/shop/exception/EntityNotFoundException.kt
package io.github.artemy.osipov.shop.exception

import java.lang.RuntimeException
import kotlin.reflect.KClass

class EntityNotFoundException(val targetClass: KClass<*>, val id: Any) : RuntimeException(
    generateMessage(
        targetClass, id
    )
) {
    private companion object {
        fun generateMessage(targetClass: KClass<*>, id: Any): String {
            return String.format(
                "Entity '%s' with id '%s' not exists",
                targetClass.simpleName, id
            )
        }
    }
}